bin/install.sh

```diff
@@ -210,19 +226,22 @@ function interactiveInstall() {
 }
 
 function unattendedInstall() {
-  if [[ -n "${INSTALL_SCANNERS}" ]]; then
-    print "Starting to install scanners..."
-    installResources "$BASE_DIR/scanners" "default" True
+  if [[ -n "${INSTALL_DEMO_APPS}" ]]; then
+    print "Starting to install 'demo-apps' into namespace '$SCB_DEMO_NAMESPACE' ..."
+    kubectl create namespace "$SCB_DEMO_NAMESPACE" || print "Namespace '$SCB_DEMO_NAMESPACE' already exists or could not be created.. "
+    installResources "$BASE_DIR/demo-apps" "$SCB_DEMO_NAMESPACE" True
   fi
 
-  if [[ -n "${INSTALL_DEMO_APPS}" ]]; then
-    print "Starting to install demo-apps..."
-    installResources "$BASE_DIR/demo-apps" "default" True
+  if [[ -n "${INSTALL_SCANNERS}" ]]; then
+    print "Starting to install 'scanners' into namespace '$SCB_NAMESPACE' ..."
+    kubectl create namespace "$SCB_NAMESPACE" || print "Namespace '$SCB_NAMESPACE' already exists or could not be created.. "
+    installResources "$BASE_DIR/scanners" "$SCB_NAMESPACE" True
```

Comment:
Using `True` in Bash is completely wrong! There is no such boolean type in Bash. The problem is: This is simply a string `"True"`, nothing more (not to confuse w/ `true` which is the binary `/usr/bin/true` which simply always exits with 0). So checking `$foo=True` may lead to errors in various ways: This will not match if $foo is "true". It will fail if there is a binary named "True" in the PATH of the executing shell and so on.

So the pattern to circumvent missing booleans in Bash is as followed:

- We agree on the assumption that boolean false is always represented as empty string (`''`).
- We agress on the assumption that boolean true is always represented as **non** empty string. By convention we use the string "true", but in fact it does not really matter. It's only to not confusing others.
- We always check for (non) emptyness. E.g. `[[ -n "${INSTALL_DEMO_APPS}" ]]` (evaluates to true if `$INSTALL_DEMO_APPS` is not empty. To test the other way use `-z`.

You can see examples of this pattern e.g. for the CLI flag `--demo-apps`.

Reply:
Thx for your explanation, didn't know that point! 

But i'm a bit confused here because the code block you mentioned here is exactly the same as your example: `[[ -n "${INSTALL_DEMO_APPS}" ]]` =  `if [[ -n "${INSTALL_SCANNERS}" ]]; then` 🤔

We can agree on the assumption but the result (codeblock) will be the same als already is, or did i miss something here?

Reply:
The point here for this particular snippet is not the test itself, but the content of the tested variable. You assigned `True` which is accidentally converted to the String `"True"` and so this works by accident. This pattern in Bash consist of two main parts:

1. The value used to represent a boolean (empty vs non-empty string).
2. The test of this value with -n or -z.

The second part is done right here. But the first one was not.

Reply:
I commented the places above: Where you use literal `True` or `False` you should use `"true"` or `""`.